### PR TITLE
Feat(eos_designs): Add platform settings for 7368X4

### LINKED ISSUE #1680: Add platform support for 7368
### Enhancement summary

This platform is used as spines in a project:
https://www.arista.com/assets/data/pdf/Datasheets/7368X4-Datasheet.pdf

But we do not have it as an option in the platform list:
https://avd.sh/en/latest/roles/eos_designs/doc/platform-settings.html

### Which component of AVD is impacted

eos_designs

### Use case example

7368 used as the spines for a deployment.

### Describe the solution you would like

Note this is a modular switch (although only has 1 sup). Mgmt port "Management0" can be used as other modular switches.

### Describe alternatives you have considered

Set platform as 7500R3 for example instead, but timers might not apply.

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Add support for platform specific default variable (management interface and reload delay counters for mlag and non-mlag) for 7368X4 series.

## Related Issue(s)

Fixes #1680

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Added an entry for platform series 7368X4 in the `roles/eos_designs/defaults/main/main.yml` file reflecting the platform defaulrs:

```
  - platforms: [ '7368X4']
    management_interface: Management0
    reload_delay:
      mlag: 300
      non_mlag: 330
```

* 7368X4 is a modular chassis and its management interface is Management0
* The HW platform timers for 7368X4 are mlag 300ms and non-mlag 300ms. In order to follow the EVPN guide best practices, the non-mlag timer is set by default for this platform to 330ms.
* Test added in molecule to generate the config to verify both settings
* Documentation updated

## How to test

Molecule test added to generate two configurations for platform 7368X4 type to verify both management interface and reload delay counters

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/defaults/main/main.yml [hunk 1]
```diff
@@ -111,6 +111,11 @@ platform_settings:
     reload_delay:
       mlag: 900
       non_mlag: 1020
+  - platforms: [ '7368X4']
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/defaults/main/main.yml]: Space needed at the end for consistency. **[ '7368X4' ]**

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/TESTS_PLATFORM.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  node_groups:
+    TESTS_PLATFORM_7368X4:
+      mlag: true
+      mgmt_gateway: 42.42.42.1
+      platform: 7368X4
+      uplink_interfaces:
+      mlag_interfaces: [ Ethernet3 ]
+      mlag_peer_vlan: 4091
+      mlag_peer_ipv4_pool: 192.168.255.0/24
+      rack:
```
Comment (tgodaA) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/TESTS_PLATFORM.yml]: Any reason to keep this blank. Please remove this if not required.

File: ansible_collections/arista/avd/roles/eos_designs/defaults/main/main.yml (hunk 1 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_designs/defaults/main/main.yml]: interesting - it passed the pre-commit check like this. Fixed in next commit

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/TESTS_PLATFORM.yml (hunk 2 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/TESTS_PLATFORM.yml]: Removed it.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/hosts.yml [hunk 3]
```diff
@@ -85,6 +86,12 @@ all:
                     mgmt_interface_fabric:
                     mgmt_interface_host:
                     mgmt_interface_platform:
+            TESTS_PLATFORM:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/hosts.yml]: do we need all this to test a new platform? Normally we just assign the platform to one of the existing devices in any scenario

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/hosts.yml]: it is needed to check for MLAG counters as I need a group of 2 hosts to generate the mlag config - I tried to include it in the L2Leaf under TESTS but could not succeed - happy to change if there is a way. The advantage of cretaing this other node group is that it makes it possible to get some group_vars and so to generate smaller configs.

Of course happy to change it if there is a better way

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/hosts.yml]: Following our discussions just pushed a new commit leveraging DC1-CL1A to test for the new platform variables hence reducing the new amount of generated configruation

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**